realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -263,6 +263,27 @@ public void renameColumn(long columnIndex, String newName) {
     }
 
     /**
+     * When a {@link Table} is renamed, it is necessary to set the {@link Table#PRIMARY_KEY_CLASS_COLUMN_INDEX}
+     * entry of its primary Key table with a new class name to transfer the primary key.
+     *
+     * @param oldTableName the old table name to search its primary key.
+     * @param newTableName the new table name to reset the primary key
+     */
+    public void resetPrimaryKeyForRenamedClass(String oldTableName, String newTableName) {
+        // make sure the new table is suitable for column
+        verifyColumnName(newTableName);
+        String newClassName = tableNameToClassName(newTableName);
+        String oldClassName = tableNameToClassName(oldTableName);
+        Table pkTable = getPrimaryKeyTable();
+        long pkRowIndex = pkTable.findFirstString(PRIMARY_KEY_CLASS_COLUMN_INDEX, oldClassName);
```

Comment:
This is implementing logic that already exists. It would be much better if you used the already existing `setPrimaryKey()` function for removing/adding the keys. That way any extra rules will automatically be enforced. It probably means this method needs to be restructured/removed though.